mesa/visualization/ModelCreator.py

```diff
@@ -0,0 +1,92 @@
+import solara
+
+from mesa.visualization.solara_viz import split_model_params
+from mesa.visualization.UserParam import Slider
+
+
+@solara.component
+def ModelCreator(model, model_cls, model_params):
+    user_params, fixed_params = split_model_params(model_params)
+    model_parameters, set_model_parameters = solara.use_state(
+        {**fixed_params, **{k: v.get("value") for k, v in user_params.items()}}
+    )
+
+    def on_change(name, value):
+        set_model_parameters({**model_parameters, name: value})
+
+    def create_model():
+        model.value = model_cls(**model_parameters)
+
+    solara.use_effect(create_model, [model_parameters])
+
+    UserInputs(user_params, on_change=on_change)
+
+
+@solara.component
+def UserInputs(user_params, on_change=None):
+    """
+    Initialize user inputs for configurable model parameters.
+    Currently supports :class:`solara.SliderInt`, :class:`solara.SliderFloat`,
+    :class:`solara.Select`, and :class:`solara.Checkbox`.
+
+    Args:
+        user_params: Dictionary with options for the input, including label,
+        min and max values, and other fields specific to the input type.
+        on_change: Function to be called with (name, value) when the value of an input changes.
+    """
+
+    for name, options in user_params.items():
+
+        def change_handler(value, name=name):
+            on_change(name, value)
+
+        if isinstance(options, Slider):
+            slider_class = (
+                solara.SliderFloat if options.is_float_slider else solara.SliderInt
+            )
+            slider_class(
+                options.label,
+                value=options.value,
+                on_value=change_handler,
+                min=options.min,
+                max=options.max,
+                step=options.step,
+            )
+            continue
+
+        # label for the input is "label" from options or name
+        label = options.get("label", name)
+        input_type = options.get("type")
+        if input_type == "SliderInt":
```

Comment:
What does this structure add to the user? Isn't it easier to just let them just the Solara components directly?

Reply:
This is already present in existing SolaraViz, and so is out of scope. The point was so that user can declaratively specify the param in a JSON, and so is serializable.

Reply:
If the scope of the PR is small, the API change can be merged sooner.

Reply:
Thanks for the context, I’m not 100% up to speed with the current visualization.